### PR TITLE
Add optional LogCallback to InteropFileStream

### PR DESCRIPTION
**Pull request by DNF-SaS**
Provide progress-information when stream is consumed by 3rd-party logic without progress-callback capability.

### CODE REVIEW

**Comment by Tewr** [comment]
This looks nice. I'll merge it if you fix the formatting, indenting is inconsistent.

**Comment by DNF-SaS** [comment]
Yes. I think we should include some .editorconfig-Data..

File: src/FileReaderComponent/FileReaderJsInterop.cs [hunk 1]
```diff
@@ -142,7 +144,8 @@ public override void Flush()
             {
                 ThrowIfDisposed();
                 Console.WriteLine($"{nameof(InteropFileStream)}.{nameof(ReadAsync)}({nameof(buffer)}=byte[{buffer.Length}], {nameof(offset)}={offset}, {nameof(count)}={count})");
-                var bytesRead = await FileReaderJsInterop.ReadFileAsync(fileRef, buffer, Position + offset, count, cancellationToken);
+	            this.logCallback?.Invoke(this.Position, offset, this.Length);
+				var bytesRead = await FileReaderJsInterop.ReadFileAsync(fileRef, buffer, Position + offset, count, cancellationToken);
```
Comment (Tewr) [inline src/FileReaderComponent/FileReaderJsInterop.cs:148]: fix indentation here

**Comment by Tewr** [comment]
In general I like the idea, but I'd prefer naming it for what it is, rather than your intended purpose. It should thus be named onReadBytes or onReadChunk. And naming it that way implies that it's an event, which are normally passed to the resulting class rather than as a parameter. 

API-wise, I think it clutters up the params, and I'd prefer returning a subclass of Stream which exposes the event. I'd prefer returning an interface or something but the stream class has always been a bit ugly in that sense, so a subclass would have to do.

Something like InteropStream inheriting from ProgressStream, and ProgressStream inheriting from Stream. ProgressStream would implement the event. And ProgressStream would be the type returned from all the methods.

What are your thoughts on this?

**Comment by DNF-SaS** [comment]
I totally agree, that implementing Progress via inheritance is the better way. It's been just some quick'n'dirty implementation for making log-callbacks available when 3rd-party is consuming the stream.

**Comment by Tewr** [comment]
On a second though this could be avoided all together by just simply wrapping the stream. It doesnt have to be part of the API. I've done wrappers a few times in the past, to provide UI download reporting etc - [here](http://mel-green.com/Uploads/ProgressStream.cs) is an example of someone doing it. That way it is up to the consumer to wrap it if he wants. If I manage to find the time to clean this project up enough to make it a nuget package, I'd like the API as clean and simple as possible. 

A wrapper implemetation could be provided later on as an extension method though.

**Comment by DNF-SaS** [comment]
Maybe its the best to pass an optional Factory-Delegate into the InnerStream.
So the devs can decide on their own and it stays simple.

**Pull request closed without merging**